server/lib/database.js

```diff
@@ -47,6 +47,18 @@ function fetchAllPublishedRoundAnalysisBySemester(semester) {
   return RoundAnalysis.aggregate([{ $match: { semester, isPublished: true } }])
 }
 
+function fetchAllCanvasRoundAnalysesByCourseCode(courseCode) {
+  if (!courseCode) throw new Error('courseCode must be set')
+  log.debug('Fetching all round analyses for ' + courseCode + ' from Canvas')
+  return RoundAnalysis.aggregate([{ $match: { courseCode, isPublished: null, gradingDistribution: { $ne: null } } }])
```

Comment:
Great input! I will look into this and think of where the data model differ between the two sources. The main difference we know for sure is that kursinfoadmin analyses does have the `isPublished` field, among others. I think I'll add some more attribute names in the check. And add more documentation for this in the readme.

Reply:
Just a thought, could we make sure that newly added analysis from kursutveckling-integration-api have type property? So you could filter on `analysisType === 'canvas'` and then we assume that all docs without analysisType prop is from old admin tool?

Reply:
That is an easy fix! Thought of it but then thought it would feel risky to assume all other analyses weren't from canvas. But now I agree it may be the easiest and most clear solution :) We'll go for that!

Reply:
The pull request for this in `kursutveckling-integration-api` can be found here: https://github.com/KTH/kursutveckling-integration-api/pull/19

Reply:
I've updated the fetch in `database.js`. Let me know what you think! I still prefer to seperarate the two endpoints.